cluster/engine.go

```diff
@@ -1080,7 +1072,7 @@ func (e *Engine) CreateContainer(config *ContainerConfig, name string, pullImage
 	e.CheckConnectionErr(err)
 	if err != nil {
 		// If the error is other than not found, abort immediately.
-		if (err != dockerclient.ErrImageNotFound && !engineapi.IsErrImageNotFound(err)) || !pullImage {
+		if (err != testErrImageNotFound && !engineapi.IsErrImageNotFound(err)) || !pullImage {
```

Comment:
@allencloud `testErrImageNotFound` is only useful for our unit tests. Check how it's used in `engine_test.go`. It's not an error that is returned by Docker.